cuda_core/tests/test_memory.py

```diff
@@ -219,6 +219,88 @@ def test_buffer_copy_from():
     buffer_copy_from(DummyPinnedMemoryResource(device), device, check=True)
 
 
+def buffer_fill(dummy_mr: MemoryResource, device: Device, check=False):
+    stream = device.create_stream()
+
+    # Test width=1 (byte fill)
+    buffer1 = dummy_mr.allocate(size=1024)
+    buffer1.fill(0x42, width=1, stream=stream)
+    device.sync()
+
+    if check:
```

Comment:
Why are parametrizing these value checks in a test suite?  The memory sizes don't strike me as so large that these operations would be slow.